2-repositories/scala-repo-server/src/main/scala/com.snowplowanalytics.iglu.server/service/ValidationService.scala

```diff
@@ -18,135 +18,148 @@ package service
 // This project
 import actor.SchemaActor._
 import actor.ApiKeyActor._
-import util.TokenAuthenticator
 
 // Akka
 import akka.actor.ActorRef
 import akka.pattern.ask
 
+// Akka Http
+import akka.http.scaladsl.model.StatusCode
+import akka.http.scaladsl.server.{Directive1, Directives}
+import akka.http.scaladsl.model.headers.HttpChallenges
+import akka.http.scaladsl.server.AuthenticationFailedRejection.CredentialsRejected
+import akka.http.scaladsl.server.{AuthenticationFailedRejection, Route}
+import akka.http.scaladsl.settings.RoutingSettings
+
+// javax
+import javax.ws.rs.Path
+
 //Scala
 import scala.concurrent.ExecutionContext
-import scala.reflect.ClassTag
-
-// Spray
-import spray.http.StatusCodes._
-import spray.http.StatusCode
-import spray.http.MediaTypes._
-import spray.routing._
-import spray.routing.PathMatcher.Lift
 
 // Swagger
-import com.wordnik.swagger.annotations._
+import io.swagger.annotations._
+
 
 /**
- * Service to validate schemas.
- * @constructor creates a new validation service with a schema and apiKey actors
- * @param schemaActor a reference to a ``SchemaActor``
- * @param apiKeyActor a reference to a ``ApiKeyActor``
- */
-@Api(value = "/api/schemas/validate", position = 1,
-  description = "Operations dealing with schema validation")
+  * Service to validate schemas.
+  * @constructor creates a new validation service with a schema and apiKey actors
+  * @param schemaActor a reference to a ``SchemaActor``
+  * @param apiKeyActor a reference to a ``ApiKeyActor``
+  */
+@Api(value = "/api/schemas/validate", tags = Array("validation"),
+    authorizations = Array(new Authorization(value = "APIKeyHeader")), produces = "text/plain")
+@Path("/api/schemas/validate")
 class ValidationService(schemaActor: ActorRef, apiKeyActor: ActorRef)
-(implicit executionContext: ExecutionContext) extends Directives with Service {
+                       (implicit executionContext: ExecutionContext, routingSettings: RoutingSettings)
+                       extends Directives with Service {
 
   /**
-   * Creates a ``TokenAuthenticator`` to extract the apikey http header and
-   * validates it against the database.
-   */
-  val authenticator = TokenAuthenticator[(String, String)]("apikey") {
-    key => (apiKeyActor ? GetKey(key)).mapTo[Option[(String, String)]]
+    * Directive to authenticate a user.
+    */
+  def auth(key: String): Directive1[(String, String)] = {
+    val credentialsRequest = (apiKeyActor ? GetKey(key)).mapTo[Option[(String, String)]] map {
+      // if the authentication succeeds
+      case Some(t) => Right(t)
+      // if the authentication fails
+      case None => Left(AuthenticationFailedRejection(CredentialsRejected, HttpChallenges.basic("")))
```

Comment:
Shouldn't it be something like `CredentialsRejected, HttpChallenges.basic("Iglu Registry")`?

Reply:
Done